lib/graphiti/hash_renderer.rb

```diff
@@ -1,39 +1,63 @@
 module Graphiti
   module SerializableHash
-    def to_hash(fields: nil, include: {})
+    def to_hash(fields: nil, include: {}, name_chain: [], graphql: false)
       {}.tap do |hash|
         fields_list = fields[jsonapi_type] if fields
+        if fields_list.nil?
+          fields_list = fields[name_chain.join(".").to_sym]
+        end
+
+        # polymorphic resources
+        if @resource.respond_to?(:type) && @resource.type != jsonapi_type
+          if fields[@resource.type]
+            fields_list ||= []
+            fields_list |= fields[@resource.type]
+          end
+        end
+
         attrs = requested_attributes(fields_list).each_with_object({}) { |(k, v), h|
-          h[k] = instance_eval(&v)
+          name = graphql ? k.to_s.camelize(:lower).to_sym : k
+          h[name] = instance_eval(&v)
         }
         rels = @_relationships.select { |k, v| !!include[k] }
         rels.each_with_object({}) do |(k, v), h|
           serializers = v.send(:resources)
-          attrs[k] = if serializers.is_a?(Array)
+          name = graphql ? k.to_s.camelize(:lower) : k
+          name_chain << k unless name_chain.last == k
+          attrs[name.to_sym] = if serializers.is_a?(Array)
             serializers.map do |rr| # use private method to avoid array casting
-              rr.to_hash(fields: fields, include: include[k])
+              rr.to_hash(fields: fields, include: include[k], graphql: graphql, name_chain: name_chain)
```

Comment:
Should this be `name_chain.dup` to avoid accidentally modifying the same thing across multiple loops?

Reply:
Will work on adding a spec for this in either case, good thought